### PR TITLE
Speedup DateLUT initialization

### PR DESCRIPTION
**Pull request by davenger**
### Changelog category (leave one):
- Performance Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Speedup initialization of DateTime lookup tables for time zones. This should reduce startup/connect time of clickhouse client especially in debug build as it is rather heavy.

### CODE REVIEW

**Comment by robot-clickhouse-ci-2** [comment]
This is an automated comment for commit 8243b624e91492e440400254b0c6be117de65c2e with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/51347/8243b624e91492e440400254b0c6be117de65c2e/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

File: src/Common/DateLUTImpl.cpp [hunk 1]
```diff
@@ -103,38 +141,14 @@ DateLUTImpl::DateLUTImpl(const std::string & time_zone_)
...
-                values.amount_of_offset_change_value = -24 * 4;
-
-            /// We don't support cases when time change results in switching to previous day.
-            /// Shift the point of time change later.
-            if (values.time_at_offset_change_value + values.amount_of_offset_change_value < 0)
-                values.time_at_offset_change_value = -values.amount_of_offset_change_value;
-        }
-
         /// Going to next day.
+        start_of_day_time_point += std::chrono::hours(24);
```
Comment (alexey-milovidov) [inline src/Common/DateLUTImpl.cpp:151]: I've got the idea, but this variable name is weird. Let's rename it to "start_of_day_time_point_if_no_transitions" or "start_of_day_time_point_usually" or similar.

Comment (alexey-milovidov) [reply src/Common/DateLUTImpl.cpp:151]: + writing a small comment will help.

**Review by alexey-milovidov — APPROVED** [review]

File: src/Common/DateLUTImpl.cpp (hunk 1 above)
Comment (davenger) [reply src/Common/DateLUTImpl.cpp]: done

**Pull request merged**